Lib/multiprocessing/popen_spawn_win32.py

```diff
@@ -87,12 +89,21 @@ def __init__(self, process_obj):
                                            (self.sentinel, int(rhandle)))
 
             # send information to child
-            set_spawning_popen(self)
-            try:
-                reduction.dump(prep_data, to_child)
-                reduction.dump(process_obj, to_child)
-            finally:
-                set_spawning_popen(None)
+            # bpo-42178: write to BytesIO to check for pickler exceptions
+            with io.BytesIO() as f:
+                set_spawning_popen(self)
+                try:
+                    reduction.dump(prep_data, f)
+                    reduction.dump(process_obj, f)
+                except:
+                    _winapi.TerminateProcess(self._handle)
```

Comment:
You need the exit code: `_winapi.TerminateProcess(self._handle, 1)`. Also add a `raise` statement to propagate the exception to the caller.